tests/test_web.py

```diff
@@ -225,16 +227,109 @@ def test_smtp_starttls_env_flag_must_be_true_boolean(environment, http, server,
     indirect=["environment"]
 )
 def test_email_body_was_sent_with_expected_body(
-        environment, http, server, email_invoice_request_body, expected_body
+    environment, http, server, mock_render, email_invoice_request_body, expected_body
 ):
     http.post("/invoice", json=email_invoice_request_body)
 
-    email_content = server.sendmail.call_args.args[2]
+    email_content = server.sendmail.call_args.args[MESSAGE_ARGUMENT]
     assert expected_body in email_content
 
 
+@pytest.mark.parametrize(
+    "environment",
+    [
+        (
+            {
+                "INVOICE_UTILS_MAIL_SUBJECT": "test subject",
+                "INVOICE_UTILS_BODY_TEMPLATE_PATH": "test_template.html",
+                "INVOICE_UTILS_SENDER_EMAIL": "test@email.com"
+            }
+        )
+    ],
+    indirect=["environment"]
+)
+def test_sendmail_was_called_with_pdf_attachment(
```

Comment:
I see that we're testing whether sendmail is called. How about adding a test when sendmail fails with an `Exception`.

Reply:
Yes, the test at line 127 checks this, any error raised within the smtplib context is handeled with a specific message.